### PR TITLE
Upgrade Dotty Reference to 3.0.0

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

**Comment by smarter** [comment]
fs2 fails (I don't see how this could be related to this change so it's likely flakey?) /cc @mpilquist:
```scala
[error] ==> X fs2.StreamCombinatorsSuite.groupWithin - accumulation and splitting  0.481s munit.ComparisonFailException: /__w/dotty/dotty/community-build/community-projects/fs2/core/shared/src/test/scala/fs2/StreamCombinatorsSuite.scala:851
[error] 850:        .toList
[error] 851:        .assertEquals(expected)
[error] 852:    }
[error] values are not the same
[error] => Obtained
[error] List(
[error]   List(
[error]     1,
[error]     2,
[error]     3
[error]   ),
[error]   List(
[error]     4
[error]   ),
[error]   List(
[error]     5,
[error]     6,
[error]     7,
[error]     8,
[error]     9
[error]   ),
[error]   List(
[error]     10,
[error]     11,
[error]     12,
[error]     13,
[error]     14
[error]   ),
[error]   List(
[error]     15,
[error]     16,
[error]     17,
[error]     18,
[error]     19
[error]   ),
[error]   List(
[error]     20,
[error]     21,
[error]     22
[error]   )
[error] )
[error] => Diff (- obtained, + expected)
[error]    List(
[error] -    4
[error] -  ),
[error] -  List(
[error] +    4,
[error]      5,
[error]      7,
[error] -    8,
[error] -    9
[error] +    8
[error]    ),
[error]    List(
[error] +    9,
[error]      10,
[error]      12,
[error] -    13,
[error] -    14
[error] +    13
[error]    ),
[error]    List(
[error] +    14,
[error]      15,
[error]      17,
[error] -    18,
[error] -    19
[error] +    18
[error]    ),
[error]    List(
[error] +    19,
[error]      20,
[error]     at munit.FunSuite.assertEquals(FunSuite.scala:11)
[error]     at munit.CatsEffectAssertions.assertIO$$anonfun$2$$anonfun$1(CatsEffectAssertions.scala:49)
[error]     at cats.effect.IO$.apply$$anonfun$1(IO.scala:806)
[error]     at cats.effect.IOFiber.runLoop(IOFiber.scala:270)
[error]     at cats.effect.IOFiber.asyncContinueSuccessfulR(IOFiber.scala:1132)
[error]     at cats.effect.IOFiber.run(IOFiber.scala:126)
[error]     at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:384)
[error] Failed: Total 1447, Failed 1, Errors 0, Passed 1428, Skipped 18
[error] Failed tests:
[error] 	fs2.StreamCombinatorsSuite
```

**Comment by mpilquist** [comment]
Definitely an intermittent failure in fs2 test suite. I’ll mark it flaky later which will prevent it from causing a build failure.

**Review by smarter — CHANGES_REQUESTED** [review]
Now that https://github.com/lampepfl/dotty/pull/12371 is merged, we should at the same time as we bump the referenceVersion bump the `previousDottyVersion`

**Comment by anatoliykmetyuk** [comment]
Rebased on `master`.

Currently `previousDottyVersion` and `previousDottyBinaryVersion` are already set to `3.0.0-RC3` – is that not correct?

**Comment by smarter** [comment]
It should be set to the latest published version (and the whitelist in project/MiMaFilters.scala should be cleared every time it's updated), so it should be 3.0.0 (which is identical to 3.0.0-RC3 so we won't clear the whitelist this time)

File: project/Build.scala [hunk 1]
```diff
@@ -81,8 +81,8 @@ object Build {
    *  set to 3.1.3. If it is going to be 3.1.0, it must be set to the latest
    *  3.0.x release.
    */
-  val previousDottyVersion = "3.0.0-RC3"
-  val previousDottyBinaryVersion = "3.0.0-RC3"
+  val previousDottyVersion = "3.0.0"
+  val previousDottyBinaryVersion = "3.0.0"
```
Comment (smarter) [inline project/Build.scala:85]: The binary version is 3 now (can just be hardcoded in the single place where it's used)

**Review by smarter — CHANGES_REQUESTED** [review]

**Pull request merged**